File: api/banking.py
# api/banking.py
# Fiat-Banking-API: EUR/USD <-> AI/COIN/AIC_LP

from core import banking as core_banking


def deposit(vm, username: str, amount: float, currency: str, target_token: str):
    """
    Fiat-Einzahlung (EUR/USD) in AI/COIN/AIC_LP.
    Externe Payment-Provider (PayPal/Stripe/Bank) liegen außerhalb dieser Funktion.
    """
    result = core_banking.fiat_deposit(username, amount, currency, target_token)

    # Optional: Blockchain-Zustand speichern
    if hasattr(vm, "blockchain") and hasattr(vm.blockchain, "save_state"):
        vm.blockchain.save_state()

    return {
        "status": "ok" if result.get("success") else "failed",
        "type": result.get("type"),
        "username": result.get("username"),
        "amount": result.get("amount"),
        "currency": result.get("currency"),
        "token": result.get("target_token")
    }


def withdraw(vm, username: str, amount: float, currency: str, source_token: str):
    """
    Fiat-Auszahlung (EUR/USD) aus AI/COIN/AIC_LP.
    """
    result = core_banking.fiat_withdraw(username, amount, currency, source_token)

    if hasattr(vm, "blockchain") and hasattr(vm.blockchain, "save_state"):
        vm.blockchain.save_state()

    return {
        "status": "ok" if result.get("success") else "failed",
        "type": result.get("type"),
        "username": result.get("username"),
        "amount": result.get("amount"),
        "currency": result.get("currency"),
        "token": result.get("source_token")
    }

File: scripts/banking.js
// scripts/banking.js
// Fiat-Banking: EUR/USD <-> AI/COIN/AIC_LP

export function initBankingUI() {
  const depositForm = document.getElementById('fiat-deposit-form');
  const withdrawForm = document.getElementById('fiat-withdraw-form');

  if (depositForm) {
    depositForm.addEventListener('submit', async (e) => {
      e.preventDefault();
      const formData = new FormData(depositForm);
      const username = formData.get('username');
      const amount = parseFloat(formData.get('amount'));
      const currency = formData.get('currency');
      const token = formData.get('token'); // AI, COIN, AIC_LP

      const res = await fetch('/api/banking/deposit', {
        method: 'POST',
        headers: {'Content-Type': 'application/json'},
        body: JSON.stringify({ username, amount, currency, target_token: token })
      });

      const data = await res.json();
      console.log('Fiat deposit result:', data);
    });
  }

  if (withdrawForm) {
    withdrawForm.addEventListener('submit', async (e) => {
      e.preventDefault();
      const formData = new FormData(withdrawForm);
      const username = formData.get('username');
      const amount = parseFloat(formData.get('amount'));
      const currency = formData.get('currency');
      const token = formData.get('token'); // AI, COIN, AIC_LP

      const res = await fetch('/api/banking/withdraw', {
        method: 'POST',
        headers: {'Content-Type': 'application/json'},
        body: JSON.stringify({ username, amount, currency, source_token: token })
      });

      const data = await res.json();
      console.log('Fiat withdraw result:', data);
    });
  }
}

